### ISSUE TITLE
isdir_or_mkdir breaks on // network export

### ISSUE THREAD

**Issue by stephenwhitmarsh** [fieldtrip/fieldtrip#1276]
Not sure how it's called, but at our institute the file server is accessed locally via: //lexport/
isdir_or_mkdir will break on this because of the double filesep.
 
- Not sure why it only now became a problem.
- I didn't expert ft_write_data to (try to) create directories for me. Personally this is unwanted (because unrequested) behaviour.

**Comment by robertoostenveld** [comment]
can you suggest a solution for the actual problem?

**Comment by stephenwhitmarsh** [comment]
My suggested fix is to remove the check, or replace it with MATLAB's isfolder (and if FALSE, break). 
If that is acceptable, I will do a PR

**Comment by robertoostenveld** [comment]
no, that is not acceptable. The `isdir_or_mkdir` function is there for a good reason, and it should be solved inside that function.

**Comment by robertoostenveld** [comment]
please search though the previous issues for `isdir_or_mkdir`, I recall it being introduced about 1-2 year ago by @nno.

**Comment by nno** [comment]
@robertoostenveld I'm not sure what you think I introduced:

```
$ git blame private/isdir_or_mkdir.m
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  1) function isdir_or_mkdir(p)
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  2)
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  3) % ISDIR_OR_MKDIR Checks that a directory exists, or if not, creates the directory and
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  4) % all its parent directories.
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  5) %
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  6) % See also FOPEN_OR_ERROR
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  7)
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  8) tok = tokenize(p, filesep);
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200  9) if isempty(tok{1})
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 10)   tok{1} = filesep;
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 11) end
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 12) for i=1:numel(tok)
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 13)   d = fullfile(tok{1:i});
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 14)   if ~isfolder(d)
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 15)     ft_notice('creating directory %s', d);
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 16)     mkdir(d);
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 17)   end
be375db9 (Robert Oostenveld 2019-05-23 17:14:48 +0200 18) end
```

In any case I will submit a PR addressing this.

**Comment by nno** [comment]
@stephenwhitmarsh can you see if PR #1279 addresses the issue?

**Comment by stephenwhitmarsh** [comment]
Hi Nikolaas,

Thanks a lot. I am running my script now using your branch. I should know
today (it's slow but easiest for me to test like this :-))

cheers,
stephen

Op vr 13 dec. 2019 om 12:21 schreef Nikolaas N. Oosterhof <
notifications@github.com>:

**Comment by stephenwhitmarsh** [comment]
@nno It works fine for me now! Thanks a lot.

**Comment by robertoostenveld** [comment]
this has been addressed with #1279
